tests/hypothesis_strategies/basic.py

```diff
@@ -3,6 +3,14 @@
 from functools import partial
 
 import hypothesis.strategies as st
+import numpy as np
 
-finite_floats = partial(st.floats, allow_infinity=False, allow_nan=False)
+from baybe.utils.numerical import DTypeFloatNumpy
+
+finite_floats = partial(
+    st.floats,
+    allow_infinity=False,
+    allow_nan=False,
+    width=32 if DTypeFloatNumpy == np.float32 else 64,
+)
```

Comment:
I think this part here perhaps mixes two properties that should rather stay independent – because we want the float32/64 control even for non-finite floats. Therefore, perhaps it makes more sense to change this into
```suggestion
dtype_floats = partial(st.floats, width=32 if DTypeFloatNumpy == np.float32 else 64)
"""A strategy producing floats of the requested dtype."""

finite_floats = partial(dtype_floats, allow_infinity=False, allow_nan=False)
```
And then really use `dtype_floats` in placed where we only want to control the type?